core/runtime.rs

```diff
@@ -811,7 +816,10 @@ impl JsRuntime {
           .expect("Expected to get promise as module evaluation result");
         let promise_global = v8::Global::new(scope, promise);
         let mut state = state_rc.borrow_mut();
-        state.pending_promise_exceptions.remove(&promise_global);
+        let removed = state.pending_promise_exceptions.remove(&promise_global);
+        if removed.is_none() {
+          state.cancelled_promise_exceptions.insert(promise_global);
```

Comment:
Actually I think this is leaky, dynamic imports which don't eventually reject are also added to and never removed from this set. @bartlomieju Any ideas?

Reply:
Though it's not that big of a deal, as data associated with that module is also permanently kept in the module graph I think.

Reply:
@nayeemrmn can you remove the promise handle from this hash map once dynamic import resolves?

Reply:
> @nayeemrmn can you remove the promise handle from this hash map once dynamic import resolves?

No, as far as the runtime infrastructure is concerned, the module is considered evaluated and done with once the pending promise is returned. Then it's just a promise in JS land. We pretty much need to do the equivalent of `.then()`ing it in rust land at this site.

Reply:
> > @nayeemrmn can you remove the promise handle from this hash map once dynamic import resolves?
> 
> No, as far as the runtime infrastructure is concerned, the module is considered evaluated and done with once the pending promise is returned. Then it's just a promise in JS land. We pretty much need to do the equivalent of `.then()`ing it in rust land at this site.

Would https://docs.rs/rusty_v8/0.19.0/rusty_v8/struct.Promise.html#method.then work?

Reply:
Discussed on discord: https://discord.com/channels/684898665143206084/713035341736706089/810923927618060308.